src/Dialogs/SmartPlaylistEditor.vala

```diff
@@ -285,13 +285,18 @@ public class Noise.SmartPlaylistEditorQuery : GLib.Object {
         field_combobox.append_text (_("Skipcount"));
         field_combobox.append_text (_("Title"));
         field_combobox.append_text (_("Year"));
+        field_combobox.append_text (_("URI"));
 
         field_combobox.set_active ((int)q.field);
         debug ("setting filed to %d\n", q.field);
         comparator_combobox.set_active ((int)q.comparator);
 
         if (needs_value (q.field)) {
-            value_entry.text = q.value.get_string ();
+            if(q.field == SmartQuery.FieldType.URI) {
+                value_entry.text = q.value.get_string ().replace("%20", " ");
```

Comment:
Is there a reason why [GLib.Uri](https://valadoc.org/glib-2.0/GLib.Uri.html) cannot be used in this case?